### PR TITLE
fix: respect pause and resume on connections

### PR DESCRIPTION
**Pull request by dignifiedquire**
@diasdavid please review and release when happy

### CODE REVIEW

**Comment by diasdavid** [comment]
I'm suspicious that without any 'unpipe', this will create errors when we have other streams writing to the conn stream after the handshake has been performed.

How tested was this upgrade with swarm and secio tests?

**Comment by dignifiedquire** [comment]
@diasdavid can you write a test for the concern you have?

**Comment by diasdavid** [comment]
The best is to link it to swarm and make sure it passes all the tests cause there is a lot of writes coming from several places

**Comment by dignifiedquire** [comment]
swarm + secio passes with this change, that's why I implemented it in the first place

File: src/dialer.js [hunk 1]
```diff
@@ -33,8 +39,8 @@ function Dialer () {
       return callback(new Error('multistream handshake has not finalized yet'))
     }
 
-    lpm.write(conn, new Buffer(protocol + '\n'))
-    lpm.read(conn, function (msgBuffer) {
+    encode.write(new Buffer(protocol + '\n'))
+    decode.once('data', function (msgBuffer) {
```
Comment (diasdavid) [inline src/dialer.js]: ah, got it. 'once' and not 'on'

**Comment by diasdavid** [comment]
@dignifiedquire released on 0.10

**Comment by dignifiedquire** [comment]
thanks :)

**Pull request merged**